src/query/parser/promql/parse.go

```diff
@@ -82,11 +84,29 @@ func (p *parseState) transformLen() int {
 	return len(p.transforms)
 }
 
-func (p *parseState) addLazyTransform(lazyOpts *block.LazyOpts) error {
-	if lazyOpts == nil {
+func (p *parseState) addLazyTransform(offset time.Duration) error {
+	// NB: if offset is <= 0, we do not apply any offsets.
+	switch {
+	case offset == 0:
+		return nil
+	case offset < 0:
+		// todo(braskin): add zap logger here.
+		log.Printf("offset must be positive, received: %v. not applying offset", offset)
```

Comment:
I think instead of this may be easier to error out? It's not really a viable use case anyway since the Prom parser will error with negative offsets

Reply:
Yeah thats what I was thinking.. glad we're on the same page.

Reply:
I don't think we should land this without a zap Logger passed in here.  Can you please find a way to propagate a logger to the `parseState`?

Reply:
Returning an error instead of logging.